sources/updater/signpost/src/state.rs

```diff
@@ -243,10 +243,14 @@ impl State {
 
     /// Reverts upgrade_to_inactive(), but **does not write to the disk**.
     ///
-    /// * Clears all bits of the inactive partition.
+    /// * Sets the inactive partition's priority to 0
+    /// * Sets the inactive partition as not successfully booted.
     /// * Restores the active partition's priority to 2
     pub fn cancel_upgrade(&mut self) {
-        self.clear_inactive();
+        let mut inactive_flags = self.gptprio(self.inactive());
+        inactive_flags.set_priority(0);
+        inactive_flags.set_successful(false);
```

Comment:
Is setting successful=false necessary?  It's seems unrelated to canceling an upgrade.  Maybe the other partition has been booted successfully..?